contracts/test/bridge/tokenBridge.ts

```diff
@@ -0,0 +1,184 @@
+import { expect } from 'chai'
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers'
+import { expectAddressEquality, loadStarkgateContract, loadProxyContract } from '../utils'
+
+describe('Test starkgate bridge with link token', function () {
+  this.timeout(TIMEOUT)
+  const networkUrl: string = (network.config as HttpNetworkConfig).url
+  let owner: Account
+  let tokenBridgeContract: StarknetContract
+  let linkTokenContract: StarknetContract
+  let deployer: SignerWithAddress
+
+  let starkNetERC20Bridge: Contract
+  let mockStarknetMessaging: Contract
+  let proxy: Contract
+  let testERC20: Contract
+  let newStarkNetERC20Bridge: Contract
+
+  before(async () => {
+    owner = await starknet.deployAccount('OpenZeppelin')
+
+    let tokenBridgeFactory = await starknet.getContractFactory(
+      '../../node_modules/@chainlink-dev/starkgate-contracts/artifacts/token_bridge.cairo',
+    )
+    tokenBridgeContract = await tokenBridgeFactory.deploy({
+      governor_address: owner.starknetContract.address,
+    })
+
+    let linkTokenFactory = await starknet.getContractFactory('link_token')
+    linkTokenContract = await linkTokenFactory.deploy({ owner: tokenBridgeContract.address })
+
+    const accounts = await ethers.getSigners()
+    deployer = accounts[0]
+
+    const contract = await loadStarkgateContract('StarknetERC20Bridge')
+    const starkNetERC20BridgeFactory = new ethers.ContractFactory(
+      contract.abi,
+      contract.bytecode,
+      deployer,
+    )
+    starkNetERC20Bridge = await starkNetERC20BridgeFactory.deploy()
+    await starkNetERC20Bridge.deployed()
+
+    const mockStarknetMessagingFactory = await ethers.getContractFactory(
+      'MockStarkNetMessaging',
+      deployer,
+    )
+    mockStarknetMessaging = await mockStarknetMessagingFactory.deploy()
+    await mockStarknetMessaging.deployed()
+
+    await starknet.devnet.loadL1MessagingContract(networkUrl, mockStarknetMessaging.address)
+
+    const testERC20Factory = await ethers.getContractFactory('TestERC20', deployer)
+    testERC20 = await testERC20Factory.deploy()
+    await testERC20.deployed()
+
+    const proxyContract = loadProxyContract('Proxy')
+    const proxyFactory = new ethers.ContractFactory(
+      proxyContract.abi,
+      proxyContract.bytecode,
+      deployer,
+    )
+    proxy = await proxyFactory.deploy(0)
+    await proxy.deployed()
+  })
+
+  describe('Test bridge from L1 to L2', function () {
+    it('Test Set and Get function for L2 token address', async () => {
+      const new_data = ethers.utils.hexConcat([
+        ethers.utils.hexZeroPad(ethers.constants.AddressZero, 32),
+        ethers.utils.hexZeroPad(testERC20.address, 32),
+        ethers.utils.hexZeroPad(mockStarknetMessaging.address, 32),
+      ])
+
+      await proxy.connect(deployer).addImplementation(starkNetERC20Bridge.address, new_data, false)
+      await proxy.connect(deployer).upgradeTo(starkNetERC20Bridge.address, new_data, false)
+    })
```

Comment:
What are we testing here exactly? Maybe you wanted to define a before hook here (that will point the proxy to the contract we're testing) and not the test (`it`).